src/client/datascience/data-viewing/jupyterVariableDataProvider.ts

```diff
@@ -53,6 +54,21 @@ export class JupyterVariableDataProvider implements IJupyterVariableDataProvider
         });
     }
 
+    // Parse a string of the form (1, 2, 3)
+    private static parseShape(shape: string) {
```

Comment:
Looks like ShapeSplitterRegEx might be doing something similar?

Reply:
Oops, missed this one. ShapeSplitterRegEx is inadequate for this use case--it only works for 2D data, and here we need all the components of the shape in a list, but the regex only generates a capture group for the last number: https://regex101.com/r/0a1nQz/1 

We also can't replace ShapeSplitterRegEx with this function, because ShapeSplitterRegEx exists to match a shape string within a Jupyter inspect result.